crates/bevy_egui/src/render.rs

```diff
@@ -0,0 +1,240 @@
+}
+
+impl Default for EguiCameraComponents {
+    fn default() -> Self {
+        // we want 0 to be "closest" and +far to be "farthest" in 2d, so we offset
+        // the camera's translation by far and use a right handed coordinate system
+        let far = 1000.0;
+        EguiCameraComponents {
+            camera: Camera {
+                name: Some(camera::EGUI_CAMERA_NAME.to_string()),
+                ..Default::default()
+            },
+            orthographic_projection: OrthographicProjection {
+                far,
+                window_origin: WindowOrigin::BottomLeft,
+                ..Default::default()
+            },
+            visible_entities: Default::default(),
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, far - 0.1)),
+            global_transform: Default::default(),
+        }
+    }
+}
+
+pub const EGUI_PIPELINE_HANDLE: Handle<PipelineDescriptor> =
+    // No clue where the hell this number comes from, but I changed a few of them to be slightly different
+    Handle::from_u128(323432602125399287835112542539754486265);
+
+pub fn build_egui_pipeline(shaders: &mut Assets<Shader>) -> PipelineDescriptor {
+    PipelineDescriptor {
+        index_format: IndexFormat::Uint16,
+        primitive_topology: PrimitiveTopology::TriangleList,
+
+        rasterization_state: Some(RasterizationStateDescriptor {
+            front_face: FrontFace::Ccw,
+            cull_mode: CullMode::None,
+            depth_bias: 0,
+            depth_bias_slope_scale: 0.0,
+            depth_bias_clamp: 0.0,
+            clamp_depth: false,
+        }),
+        depth_stencil_state: Some(DepthStencilStateDescriptor {
+            format: TextureFormat::Depth32Float,
+            depth_write_enabled: false,
+            depth_compare: CompareFunction::Always,
+            stencil: StencilStateDescriptor {
+                front: StencilStateFaceDescriptor::IGNORE,
+                back: StencilStateFaceDescriptor::IGNORE,
+                read_mask: 0,
+                write_mask: 0,
+            },
+        }),
+        color_states: vec![ColorStateDescriptor {
+            format: TextureFormat::Bgra8UnormSrgb,
+            color_blend: BlendDescriptor {
+                src_factor: BlendFactor::SrcAlpha,
+                dst_factor: BlendFactor::OneMinusSrcAlpha,
+                operation: BlendOperation::Add,
+            },
+            alpha_blend: BlendDescriptor {
+                src_factor: BlendFactor::OneMinusDstAlpha,
+                dst_factor: BlendFactor::One,
+                // src_factor: BlendFactor::One,
+                // dst_factor: BlendFactor::OneMinusSrcAlpha,
+                operation: BlendOperation::Add,
+            },
+            write_mask: ColorWrite::ALL,
+        }],
+        ..PipelineDescriptor::new(ShaderStages {
+            vertex: shaders.add(Shader::from_glsl(
+                ShaderStage::Vertex,
+                include_str!("egui.vert"),
```

Comment:
Not a change request, just a question:

Can shaders be loaded through the asset system (and thus hot reloaded)?

Reply:
Uhh... Maybe? All the shaders in bevy are statically included though. So I'm guessing not yet probably?